docs/metrics/difficulty-score.md

```diff
@@ -0,0 +1,215 @@
+| --- | --- | --- | --- | --- |
+| `1` | `0.3` | `0.3` | `0.3` | `0.30` |
+| `2` | `0.1` | **`0.9`** | `0.1` | **`0.37`** |
+| `3` | `0.4` | **`0.2`** | `0.6` | **`0.4`** |
+
+From the table above, we see that the 2nd datapoint performs very poorly on the 2nd model with a difficulty score of
+`0.9`, while the 3rd datapoint has `0.2` just underneath. However, the computed `difficulty_score` values indicate
+that the 3rd datapoint is repeatedly the most challenging datapoint for the models based on the defined
+quality standard.
+
+### Difficulty Scores for Task Metrics
+
+Difficulty scores for task metrics are aggregate metrics that do not offer datapoint-level details
+of performance. However, the information provided to an aggregate metric is sufficient in establishing
+difficulty scores at the datapoint level, similar to datapoint-level `inference_time` or `cost`.
+
+#### Binary Classification and Regression
+
+The difficulty score is the absolute error between the model result and the ground truth value.
+
+Suppose in a binary classification problem, a model's inference is binarized by the threshold of `0.5`,
+so the positive class would be defined by values `0.5` to `1.0`, and values of the negative class would
+be from `0.0` to `0.5`.
+
+| id | ground_truth | model_result | delta | difficulty_contribution |
+| --- | --- | --- | --- | --- |
+| `1` | `1`| `0.01` | `0.99` | `1.00` |
+| `2` | `1`| `0.49` | `0.51` | `0.51` |
+| `3` | `1`| `0.50` | `0.50` | `0.50` |
+| `4` | `1`| `0.80` | `0.20` | `0.19` |
+| `5` | `0`| `0.01` | `0.01` | `0.00` |
+| `6` | `0`| `0.49` | `0.49` | `0.49` |
+| `7` | `0`| `0.50` | `0.50` | `0.50` |
+| `8` | `0`| `0.80` | `0.80` | `0.81` |
+
+In the case of regression problems, difficulty can be measured in a similar way using the magnitude of the
+difference between the ground truth and the model result.
+
+| id | ground_truth | model_result | delta | difficulty_contribution |
+| --- | --- | --- | --- | --- |
+| `1` | `1`| `1` | `0` | `0.00` |
+| `2` | `2`| `1` | `1` | `0.08` |
+| `3` | `3`| `2` | `1` | `0.08` |
+| `4` | `4`| `3` | `1` | `0.08` |
+| `5` | `5`| `5` | `0` | `0.00` |
+| `6` | `6`| `8` | `2` | `0.15` |
+| `7` | `7`| `13` | `6` | `0.46` |
+| `8` | `8`| `21` | `13` | `1.00` |
+
+The greater the distance the model result is from the ground truth, the greater the difficulty of that
+datapoint. These normalized `delta` column, called a `difficulty_contribution` column, becomes another
+column in step 4 of the example above which is parallel to `cost` or `recall`.
+
+| id | recall | difficulty_contribution | ... | difficulty_score |
+| --- | --- | --- | --- | --- |
+| `1` | `1.0`| `0.2` | ... | ... |
+| `2` | `0.5`| `0.9` | ... | ... |
+
+Then, it can be involved in the computation of the overall `difficulty_score` for the datapoint.
+
+#### Multiclass Classification
+
+The `delta` column for a datapoint of a multiclass classification task is simply the number of times a model
```

Comment:
What does it mean by `number of times`?  Isn't is just a correct vs. incorrect classification per datapoint?